postgres/tests/test_statements.py

```diff
@@ -627,6 +627,81 @@ def obfuscate_sql(query, options=None):
     assert metric['dd_commands'] == expected_metadata_payload['commands']
 
 
+@pytest.mark.parametrize("pg_stat_statements_view", ["pg_stat_statements", "datadog.pg_stat_statements()"])
+@pytest.mark.parametrize(
+    "set_reported_hostname,expected_reported_hostname",
+    [
+        (True, 'fred_hostname_override'),
+        (False, 'stubbed.hostname'),
+    ],
+)
+def test_statement_reported_hostname(
+    aggregator,
+    integration_check,
+    dbm_instance,
+    datadog_agent,
+    pg_stat_statements_view,
+    set_reported_hostname,
+    expected_reported_hostname,
+):
+    dbm_instance['pg_stat_statements_view'] = pg_stat_statements_view
+    dbm_instance['query_samples'] = {'enabled': True, 'run_sync': True, 'collection_interval': 0.1}
+    dbm_instance['query_metrics'] = {'enabled': True, 'run_sync': True, 'collection_interval': 0.1}
+    if set_reported_hostname:
+        dbm_instance['reported_hostname'] = expected_reported_hostname
+
+    # If the query changes, the query_signatures for both will need to be updated as well.
+    query = "SELECT city FROM persons WHERE city = 'hello'"
```

Comment:
The only setup we need for this test is the minimum amount required that produces a valid payload of each DBM data type. I think we can achieve that by just running the check twice and all data types should show up based purely on the built-in queries the agent already runs? 

That would help avoid duplicating the more involved test setup logic from the other tests.